tools/cbmc/proofs/utility/memory_assignments.c

```diff
@@ -0,0 +1,25 @@
+#define ensure_memory_is_valid( px ) (px != NULL)
+
+/* Implementation of safe malloc */
+void *safeMalloc(size_t xWantedSize) {
+	if(xWantedSize == 0) {
```

Comment:
I'm not sure this is correct. https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations

Reply:
I have added a comment.